### PR TITLE
Remove deprecated shared library feature and go back to scratch

### PR DESCRIPTION
**Pull request by tsandall**
It's always nice to remove some dependencies. In this case, we deprecated the shared plugin support a little while ago (v0.14.0) and now it's time to remove it completely. This PR removes the code (but keeping plugin registration test coverage), switches the build to disable cgo, and updates the normal docker image to use scratch as base.

### CODE REVIEW

**Comment by tsandall** [comment]
Weird failure. Fixed.

**Comment by tsandall** [comment]
@patrick-east PTAL. I snuck in a fix for the dirty working copy issue we saw in the wasm/unicode PR.

File: build/run-wasm-rego-tests.sh [hunk 1]
```diff
@@ -44,7 +44,6 @@ function generate_testcases {
     purge_testgen_container
     docker run \
         --name $TESTGEN_CONTAINER_NAME \
-        -u $(id -u):$(id -g) \
```
Comment (patrick-east) [inline build/run-wasm-rego-tests.sh:47]: Wont this break linux users where the generated files will be owned by root? (osx does special user permission mapping with the shared files into the docker vm)

Comment (tsandall) [reply build/run-wasm-rego-tests.sh:47]: ah, I didn't realize that Docker for Mac was doing some magic. OK, I'll figure something out.

**Comment by tsandall** [comment]
@patrick-east fixed -- the `go install` was the problem.

**Pull request merged**